### DISCUSSION TITLE
Scale when using native resolution

### DISCUSSION THREAD

**Discussion by webysther** [kasmtech/KasmVNC#213] [Q&A] [answered]
Hello, when I'm setting the native resolution in the KasmVNC client configuration, I came across a scenario: the application scale on a 4k monitor is 1:1, typically the scale is adjusted (Windows/Linux/Mac) to something between 125-200% depending on user preference. Is there a way to achieve this scale without reducing the resolution? I've only achieved an impressive image quality this way, so I'd like to maintain it and if possible, only change the scale. I've looked at the documentation and the available code here but haven't found anything related.

![image](https://github.com/kasmtech/KasmVNC/assets/750007/da287e16-7d53-47b3-afe5-d7b415f55c09)

**Comment by webysther** [comment, marked as answer]
- Enable debug
- Adjust the zoom and see what resolution is send

**Comment by mmcclaskey** [comment]
If you use your browser zoom, you are effectively undoing the benefits of using the native resolution feature of KasmVnc. Instead you should use your Linux desktop environment’s scaling features, just like you do on Windows or MacOS, as you pointed out. When the OS does scaling, it looks much better because fonts and other primitives are rendered at full resolution. When KasmVNC or your browser does scaling, it is done on the end resulting image using basic image scaling algorithms. So if you want the best possible picture quality, set Kasm to native resolution and keep the browser at 100% zoom and then adjust your Linux DE’s zoom/scaling.

**Reply by webysther** [reply]
Hi @mmcclaskey, I have a 4k monitor and with zoom to 80% I got the ideal size to work with the tools inside LinuxServer images, normally is one app only and I don't know where to change the linux de zoom/scaling (I will link this discussion in a issue on lsio) but one interesting point, with 4k in native resolution the experience slow down a bit but with zoom to 80% I got on debug '2856x14??' and is sharp enough to me. My problem is because I get old, 4k without zoom/scaling is too small :)

**Reply by webysther** [reply]
I changed the files `/etc/s6-overlay/s6-rc.d/svc-kasmvnc/run` and `/usr/local/etc/kasmvnc/kasmvnc.yaml` but no luck

**Reply by webysther** [reply]
Default (too big to work on photos):
![image](https://github.com/kasmtech/KasmVNC/assets/750007/8eceb060-cb91-40fc-a35b-40acd97a1b2b)

Native (too small to see anything):
![image](https://github.com/kasmtech/KasmVNC/assets/750007/c6f7a532-a466-442a-9028-eea2a410eef9)

Zoom 80% (sharp and good to work and see):
![image](https://github.com/kasmtech/KasmVNC/assets/750007/0773c527-f920-4ac6-8070-cbffe11180a0)
